src/code/views/tabbed-panel-view.tsx

```diff
@@ -37,46 +41,37 @@ class TabbedPanelTabView extends React.Component<TabbedPanelTabViewProps, Tabbed
     if (this.props.selected) {
       classname += " tab-selected";
     }
+    if (this.props.divider) {
+      classname += " tab-divider";
+    }
     return <li className={classname} onClick={this.handleClicked}>{this.props.label}</li>;
   }
 
   private handleClicked = (e) => {
     e.preventDefault();
+    if (this.props.divider) {
+      return;
+    }
     return this.props.onSelected(this.props.index);
   }
 }
 
 interface TabbedPanelViewProps {
-  selectedTabIndex?: number;
-  onTabSelected?: (index: number) => void;
+  selectedTabIndex: number;
+  onTabSelected: (index: number) => void;
   clientClass?: string;
   onRenderBelowTabsComponent?: () => any; // TODO: get concrete type
   tabs: TabInfo[];
 }
 
-interface TabbedPanelViewState {
-  selectedTabIndex: number;
-}
-
-export class TabbedPanelView extends React.Component<TabbedPanelViewProps, TabbedPanelViewState> {
+export class TabbedPanelView extends React.Component<TabbedPanelViewProps> {
 
   public static displayName = "TabbedPanelView";
 
   public static Tab(settings?: TabInfoSettings) {
     return new TabInfo(settings);
   }
 
-  constructor(props: TabbedPanelViewProps) {
```

Comment:
Not sure why we where tracking our own state version of the prop so this change seems like a good idea.  I'll be curious if this exposes some other issue on another use of the tabbed panel where we needed to track state for some reason.

Reply:
The only other use of this view was the relationship panel, which used the props-version of the selected tab. Yeah, it seemed like a good opportunity to fix an anti-pattern.